bx_django_utils/admin_utils/log_entry.py

```diff
@@ -0,0 +1,100 @@
+"""
+Helper functions around Django's admin LogEntry model.
+"""
+
+import json
+
+from django.contrib.admin.models import ACTION_FLAG_CHOICES, LogEntry
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.db.models import QuerySet
+
+ACTION_FLAG_DICT = dict(ACTION_FLAG_CHOICES)
+
+
+def validate_action_flag(value):
+    """
+    Validate that the action flag is one of the allowed values.
+    """
+    if value not in ACTION_FLAG_DICT:
+        raise ValidationError(f'Action flag {value!r} is not one of {sorted(ACTION_FLAG_DICT.keys())}')
+
+
+def create_log_entry(
+    *,
+    user,
+    instance: models.Model,
+    action_flag,
+    change_message: str | list,
+) -> LogEntry:
+    """
+    Helper to create `LogEntry` entries for a model instance.
+    """
+    content_type = ContentType.objects.get_for_model(instance)
+    log_entry = LogEntry.objects.log_action(
+        user_id=user.id,
+        content_type_id=content_type.pk,
+        object_id=instance.pk,
+        object_repr=str(instance),
+        action_flag=action_flag,
+        change_message=change_message,
+    )
+    log_entry.full_clean()
+    return log_entry
+
+
+def get_change_message_strings(queryset: QuerySet | None = None) -> list[str]:
+    """
+    Get `LogEntry` change messages as plain strings build by `LogEntry.get_change_message()`.
+    """
+    if queryset is None:
+        queryset = LogEntry.objects.order_by('action_time').all()
```

Comment:
Do we need to fetch everything here? Seems like we really only want one field, more or less, right?

Reply:
Also depends on the implementation of `.get_change_message()` that we call: It's a method that currently only access the `change_message` field.

So, i add `queryset.only('change_message')` and check the query count in tests.

Note: The `get_log_message_data()` also used primarily the `change_message` field, but if the value is not json it fallback to `str(log_entry)` that access other fields. So i don't use `.only()` here.